### PR TITLE
Password check + bonus

### PR DESCRIPTION
**Pull request by pwiatr3**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @pwiatr3

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @pwiatr3

**Comment by ziobron** [comment]
⏱ 3 XP extra for @pwiatr3 for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,60 @@
...
+bool isspecial(char ch){
+    if(isalpha(ch)){
+		return false;
+    }
+	if(isdigit(ch)){
+		return false;
+    }
+	else{
+		return true;
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:12]: Add space after `if`, `for`, `while` etc...

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,60 @@
...
\ No newline at end of file
+// TODO: Put implementations here
+bool isspecial(char ch){
+    if(isalpha(ch)){
+		return false;
+    }
+	if(isdigit(ch)){
+		return false;
+    }
+	else{
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:10]: Wrong indentation.

File: homework/password-check/validation.cpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:12]: When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,60 @@
...
+std::string getErrorMessage(ErrorCode errCode){
+    switch(errCode){
+        case Ok: return "Ok";
+        case PasswordNeedsAtLeastNineCharacters: return "Password needs to have at least nine characters";
+        case PasswordNeedsAtLeastOneNumber: return "Password needs to have at least one number";
+        case PasswordNeedsAtLeastOneSpecialCharacter: return "Password needs to have at least one special character";
+        case PasswordNeedsAtLeastOneUppercaseLetter: return "Password needs to have at least one uppercase letter";
+        case PasswordsDoNotMatch: return "Passwords do not match";
+        default: return "Wrong code";
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:24]: Once command one line

File: homework/password-check/validation.cpp [hunk 4]
```diff
@@ -1,2 +1,60 @@
...
+}
+
+bool doPasswordsMatch(std::string password1, std::string password2){
+    if(password1 == password2){
+        return true;
+    }
+    else{
+        return false;
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:33]: Add space after `if`, `for`, `while` etc...

When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

File: homework/password-check/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,60 @@
...
+
+bool doPasswordsMatch(std::string password1, std::string password2){
+    if(password1 == password2){
+        return true;
+    }
+    else{
+        return false;
+    }  
+}
+ErrorCode checkPasswordRules(std::string password){
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,60 @@
...
+        case PasswordNeedsAtLeastNineCharacters: return "Password needs to have at least nine characters";
+        case PasswordNeedsAtLeastOneNumber: return "Password needs to have at least one number";
+        case PasswordNeedsAtLeastOneSpecialCharacter: return "Password needs to have at least one special character";
+        case PasswordNeedsAtLeastOneUppercaseLetter: return "Password needs to have at least one uppercase letter";
+        case PasswordsDoNotMatch: return "Passwords do not match";
+        default: return "Wrong code";
+    }
+}
+
+bool doPasswordsMatch(std::string password1, std::string password2){
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:27]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 7]
```diff
@@ -1,2 +1,60 @@
...
+
+ErrorCode checkPassword(std::string password1, std::string password2){
+    if(doPasswordsMatch(password1, password2)){
+        return checkPasswordRules(password1);
+    }
+    else{
+        return PasswordsDoNotMatch;
+    }
+    return Ok;
+}
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:60]: Add space after `if`, `for`, `while` etc...

When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

**Review by nauka-programowania-MA — CHANGES_REQUESTED** [review]

**Pull request closed without merging**